### PR TITLE
New Redis-py Fathead

### LINKED ISSUE #435: Python: Create New Redis-py Fathead
# Redis-py Fathead

Contribute to DuckDuckGo!

screen shot 2016-10-09 at 8 57 29 pm
### What you will need to know?

Either Python (Preferrable for this mission), Ruby, Perl or JavaScript for Fatheads. You will download the Python redis client source and create an **output.txt** file containing the parsed documentation. Read more about Fatheads here: https://docs.duckduckhack.com/resources/fathead-overview.html
### The source :rocket:

https://redis-py.readthedocs.io/en/latest/
### The goal

Help _improve Python language search results_ on DuckDuckGo as part of our Programming Mission! We're trying to reach 100% Instant Answer (IA) coverage for Python-related searches by creating new Instant Answers or improving existing ones.

Some relevent examples:
- [Django Built-ins](https://github.com/duckduckgo/zeroclickinfo-fathead/tree/master/lib/fathead/django_built_ins)
- [Redis](https://github.com/duckduckgo/zeroclickinfo-fathead/tree/master/lib/fathead/redis_commands)
- [Python documentation](https://github.com/duckduckgo/zeroclickinfo-fathead/tree/master/lib/fathead/python)
### 

**Fathead Example (Django Built-ins)**

screen shot 2016-10-09 at 8 59 24 pm
### The steps
- [x] Claim this issue (Comment Below)
- [ ] Please review our [Contributing Guide](https://github.com/duckduckgo/zeroclickinfo-fathead/blob/master/CONTRIBUTING.md)
- [ ] Set up codio environment (See: https://docs.duckduckhack.com/welcome/setup-dev-environment.html) and fork this repository
- [ ] Create the Fathead
- [ ] Create a PR
- [ ] Ping @pjhampton for a review
### Questions?
- [Documentation](https://docs.duckduckhack.com/)
- [The Forum](https://forum.duckduckhack.com/c/programming/python)
- Join [DuckDuckHack Slack](https://quackslack.herokuapp.com/) 
- Ping one of the [Community Leaders](https://duck.co/help/community/community-leaders) or @pjhampton

### PR DESCRIPTION
**Pull request by vidhan13j07**
This is the Redis-py fathead discussed in #435 .

cc: @pjhampton 

Fixes #435

Forum Link: https://forum.duckduckhack.com/t/redis-py-fathead/1376
IA Page: https://duck.co/ia/view/redis_py

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/fathead/redis-py/parse.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        """
+        Iterate through the data and create the necessary output.txt file
+        """
+
+        with open('output.txt', 'w+') as output_file:
+            for data_element in self.data:
+                name = data_element.get('name').lower()
+                code = data_element.get('code')
+                paragraph = data_element.get('paragraph')
+                abstract = '<div class="prog__container">{}{}{}</div>'.format(
```
Comment (moollaza) [inline lib/fathead/redis-py/parse.py]: @vidhan13j07 please use a `<section>` tag -- we found out `<div>` creates problems

Is there still a live example with `<div>`? I thought I fixed them all...

**Comment by vidhan13j07** [comment]
@moollaza @pjhampton I have made the relevant changes.

File: lib/fathead/redis-py/parse.py (hunk 1 above)
Comment (rasikapohankar) [inline lib/fathead/redis-py/parse.py]: @moollaza the documentation has `<div class="prog__container">` as an example..

**Comment by moollaza** [comment]
@vidhan13j07 apologies, I also need you to change the directory from `redis-py` to `redis_py` -- it must be identical to the IA Page ID.

I will update it manually on my end to get this on Beta 👍

**Comment by moollaza** [comment]
@vidhan13j07 Ping! I'm just checking in to see if you're still interested in finishing this one up? We'd love to see this go live!

There's absolutely no rush, just wanted to make sure you're not stuck on anything.

**Comment by moollaza** [comment]
@vidhan13j07 I'm closing this due to inactivity. We can re-open it later if you're able to finish it up 👍

**Pull request closed without merging**